googleapiclient/http.py

```diff
@@ -1886,4 +1886,11 @@ def build_http():
         http_timeout = socket.getdefaulttimeout()
     else:
         http_timeout = DEFAULT_HTTP_TIMEOUT_SEC
-    return httplib2.Http(timeout=http_timeout)
+    http = httplib2.Http(timeout=http_timeout)
+    # 308's are used by several Google APIs (Drive, YouTube)
+    # for Resumable Uploads rather than Permanent Redirects.
+    # This asks httplib2 to exclude 308s from the status codes
+    # it treats as redirects
+    http.redirect_codes = frozenset(set(http.redirect_codes) - {308})
```

Comment:
```suggestion
    http.redirect_codes = http.redirect_codes - {308}
```

(nit)

Actually, if `http.redirect_codes` is already a `frozenset`, we can remove 308 directly, and the result will already be a frozenset. Sorry for the misleading wording in the initial comment :/

```py
>>> fs = frozenset([1, 2, 3])
>>> fs
frozenset({1, 2, 3})
>> fs - {2}
frozenset({1, 3})
```

Reply:
Thank you for taking the time to explain this! 😄